addons/crm/models/discuss_channel.py

```diff
@@ -0,0 +1,42 @@
+from odoo import models, _
+
+
+class DiscussChannel(models.Model):
+    _inherit = 'discuss.channel'
+
+    def execute_command_lead(self, **kwargs):
+        partner = self.env.user.partner_id
+        key = kwargs['body']
+        if key.strip() == '/lead':
+            msg = _('Create a new lead (/lead lead title)')
+        else:
+            lead = self._convert_visitor_to_lead(partner, key)
+            if lead:
+                msg = _('Created a new lead: %s', lead._get_html_link())
+            else:
+                msg = _('The lead can\'t be created')
+        self._send_transient_message(partner, msg)
+
+    def _get_crm_lead_vals(self, partner, key, customers):
+        return {}
```

Comment:
Can't most values already be set here?
Then livechat and whatsapp just have to fill the blank with utm sources